more_itertools/more.py

```diff
@@ -2689,3 +2691,93 @@ def map_except(function, iterable, *exceptions):
             yield function(item)
         except exceptions:
             pass
+
+
+def _sample_unweighted(iterable, k):
+    # Implementation of "Algorithm L" from the 1994 paper by Kim-Hung Li:
```

Comment:
For my own notes, the algorithm in the paper is:

![image](https://user-images.githubusercontent.com/1922815/71549553-eef9ed80-2984-11ea-8347-4224366fccde.png)